FirebaseInstallations/Source/Library/InstallationsStore/FIRInstallationsStore.m

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRInstallationsStore.h"
+
+#import <GoogleUtilities/GULUserDefaults.h>
+
+#if __has_include(<FBLPromises/FBLPromises.h>)
+#import <FBLPromises/FBLPromises.h>
+#else
+#import "FBLPromises.h"
+#endif
+
+#import "FIRInstallationsErrorUtil.h"
+#import "FIRInstallationsItem.h"
+#import "FIRInstallationsStoredItem.h"
+#import "FIRSecureStorage.h"
+
+NSString *const kFIRInstallationsStoreUserDefaultsID = @"com.firebase.FIRInstallations";
+
+@interface FIRInstallationsStore ()
+@property(nonatomic, readonly) FIRSecureStorage *secureStorage;
+@property(nonatomic, readonly, nullable) NSString *accessGroup;
+@property(nonatomic, readonly) dispatch_queue_t queue;
+@end
+
+@implementation FIRInstallationsStore
+
+- (instancetype)initWithSecureStorage:(FIRSecureStorage *)storage
+                          accessGroup:(NSString *)accessGroup {
+  self = [super init];
+  if (self) {
+    _secureStorage = storage;
+    _accessGroup = [accessGroup copy];
+    _queue = dispatch_queue_create("com.firebase.FIRInstallationsStore", DISPATCH_QUEUE_SERIAL);
+  }
+  return self;
+}
+
+- (FBLPromise<FIRInstallationsItem *> *)installationForAppID:(NSString *)appID
+                                                     appName:(NSString *)appName {
+  NSString *itemID = [FIRInstallationsItem identifierWithAppID:appID appName:appName];
+  return [self installationExistsForAppID:appID appName:appName]
+      .then(^id(id result) {
```

Comment:
The result should be the previous promise value type. The return value should be `id` because it can be either a value or an error or another promise.